structure_tests/structure_test.go

```diff
@@ -138,27 +140,59 @@ func compileAndRunRegex(regex string, base string, t *testing.T, err string, sho
 	}
 }
 
-var configFile string
-var tests StructureTest
+func (a *arrayFlags) String() string {
+	return ""
+}
 
-func init() {
-	flag.StringVar(&configFile, "config", "/workspace/structure_test.json",
-		"path to the .yaml file containing test definitions.")
-	flag.Parse()
+func (a *arrayFlags) Set(value string) error {
+	*a = append(*a, value)
+	return nil
+}
 
+func parseFile(tests *StructureTest, configFile string) {
+	var currentTests StructureTest
 	testContents, err := ioutil.ReadFile(configFile)
 	if err != nil {
 		log.Fatalf("Error reading file: %s. %s", configFile, err)
 	}
 
 	switch {
 	case strings.HasSuffix(configFile, ".json"):
-		if err := json.Unmarshal(testContents, &tests); err != nil {
+		if err := json.Unmarshal(testContents, &currentTests); err != nil {
 			log.Fatal(err)
 		}
 	case strings.HasSuffix(configFile, ".yaml"):
-		if err := yaml.Unmarshal(testContents, &tests); err != nil {
+		if err := yaml.Unmarshal(testContents, &currentTests); err != nil {
 			log.Fatal(err)
 		}
 	}
+	combineTests(tests, &currentTests)
+}
+
+func combineTests(tests *StructureTest, currentTests *StructureTest) {
+	for _, ct := range currentTests.CommandTests {
+		tests.CommandTests = append(tests.CommandTests, ct)
+	}
+	for _, fet := range currentTests.FileExistenceTests {
+		tests.FileExistenceTests = append(tests.FileExistenceTests, fet)
+	}
+	for _, fct := range currentTests.FileContentTests {
+		tests.FileContentTests = append(tests.FileContentTests, fct)
```

Comment:
I think you can just do append(tests.FileContentTests, fct...) instead of doing this in a loop.